src/coreclr/vm/typedesc.cpp

```diff
@@ -104,6 +104,31 @@ BOOL TypeDesc::IsSharedByGenericInstantiations()
     return FALSE;
 }
 
+BOOL TypeDesc::ContainsGenericVariables(BOOL methodOnly)
+{
+    if (IsGenericVariable())
+    {
+        if (!methodOnly)
+            return TRUE;
+
+        PTR_TypeVarTypeDesc pTyVar = dac_cast<PTR_TypeVarTypeDesc>(this);
+        return TypeFromToken(pTyVar->GetTypeOrMethodDef()) == mdtMethodDef;
+    }
+
+    if (HasTypeParam())
+    {
+        return GetRootTypeParam().ContainsGenericVariables(methodOnly);
```

Comment:
The similar code that was in `TypeHandle::ContainsGenericVariables` used `GetTypeParam()` instead of `GetRootTypeParam()` - is this change necessary?

Reply:
`GetRootParamType` returns the root param type that works better for nested param types. It is used in similar situations in this file.

For example, `int* [][]`. With `GetTypeParam`, we would have 3 recursive calls. With `GetRootParamType`, there won't be any recursive calls since `GetRootParamType` skips all the way to `int` immediately.